script/agglayer/DeployCustomTokensAgglayer.s.sol

```diff
@@ -0,0 +1,813 @@
+            ownerAddress,
+            bridgedVbToken,
+            bridgedUnderlyingToken,
+            agglayerBridgeAddress,
+            l1NetworkId,
+            nonMigratableBackingPercentage,
+            migrationManagerAddress
+        );
+
+        // Step 2: reinitialize2().
+        reinitializeData[1] = abi.encodeWithSelector(
+            GenericNativeConverterAgglayer.reinitialize2.selector
+        );
+
+        // Encode the call to reinitialize(bytes[]).
+        bytes memory initData = abi.encodeWithSelector(
+            GenericNativeConverterAgglayer.reinitialize.selector,
+            reinitializeData
+        );
+
+        TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
+            genericNativeConverterAgglayerImplementation,
+            proxyOwnerAddress,
+            initData
+        );
+
+        _stopBroadcast();
+
+        console.log("Generic Native Converter deployed at:", address(proxy));
+
+        return GenericNativeConverterAgglayer(address(proxy));
+    }
+
+    /// @notice Deploys a specialized vbUSDC Native Converter for chains with native USDC.
+    /// @dev This function is only called when nativeUSDC = true.
+    ///      Uses VbUsdcNativeConverterAgglayerBridgedUsdcStandard implementation which is designed
+    ///      for chains where USDC is natively issued by Circle (not bridged).
+    /// @dev Process:
+    ///      1. Computes L2 addresses for bridged vbUSDC and native USDC
+    ///      2. Fetches USDC decimals from L1
+    ///      3. Deploys TransparentUpgradeableProxy pointing to specialized implementation
+    ///      4. Initializes via reinitialize() with 2-step pattern
+    /// @return Deployed VbUsdcNativeConverterAgglayerBridgedUsdcStandard proxy (cast to GenericNativeConverterAgglayer).
+    function _deployVbUsdcNativeConverterBridgedUsdcStandard()
+        internal
+        returns (GenericNativeConverterAgglayer)
+    {
+        // Compute bridged addresses.
+        address bridgedVbUsdc = _computeBridgedAddress(vbUsdcL1);
+        address bridgedUsdc = _computeBridgedAddress(usdcL1);
+
+        // Fetch decimals from L1.
+        uint8 decimals = _fetchDecimalsFromL1(usdcL1);
+
+        console.log(
+            "Deploying vbUSDC Native Converter (Bridged USDC Standard)..."
+        );
+        console.log("Bridged vbUSDC:", bridgedVbUsdc);
+        console.log("Bridged USDC:", bridgedUsdc);
+        console.log("Decimals:", decimals);
+
+        _startBroadcast();
+
+        // Prepare reinitialize data array with 2 steps.
+        bytes[] memory reinitializeData = new bytes[](2);
+
+        // Step 1: reinitialize1(owner, customToken, underlyingToken, bridge, primaryChainId, nonMigratableBackingPercentage, migrationManager).
+        reinitializeData[0] = abi.encodeWithSelector(
+            GenericNativeConverterAgglayer.reinitialize1.selector,
+            ownerAddress,
+            bridgedVbUsdc,
```

Comment:
In case of Bridged USDC Standard, a brand new token is deployed and then custom mapped to vbUSDC from Ethereum, so the address is different from bridged vbUSDC.

https://github.com/agglayer/vault-bridge/blob/f98ff4c0a1ce3c5eb173db9ec20934002ea07be6/src/secondary-chain/agglayer/vbUSDC/bridged-usdc-standard/README.md?plain=1#L11-L12.

Reply:
Good point